cmd/curio/guidedsetup/guidedsetup.go

```diff
@@ -244,41 +246,127 @@ func complete(d *MigrationData) {
+	shell := os.Getenv("SHELL")
+	if shell == "" {
+		fmt.Println("Unable to determine the user's shell.")
+		return
+	}
+
+	var rcFile string
+	if strings.Contains(shell, "bash") {
+		rcFile = ".bashrc"
+	} else if strings.Contains(shell, "zsh") {
+		rcFile = ".zshrc"
+	} else {
+		d.say(notice, "Not adding DB variables to RC file as shell %s is not BASH or ZSH.", shell)
+		os.Exit(1)
+	}
+
+	// Get the current user's home directory
+	usr, err := user.Current()
 	if err != nil {
-		d.say(notice, "Aborting migration.", err.Error())
+		d.say(notice, "Error getting user home directory:", err)
 		os.Exit(1)
 	}
+	rcFilePath := filepath.Join(usr.HomeDir, rcFile)
 
-	args := []string{fmt.Sprintf("CURIO_DB=postgres://%s:%s@%s:%s/%s",
-		d.HarmonyCfg.Username,
-		d.HarmonyCfg.Password,
-		d.HarmonyCfg.Hosts[0],
-		d.HarmonyCfg.Port,
-		d.HarmonyCfg.Database)}
+	// Read the existing rc file content
+	file, err := os.OpenFile(rcFilePath, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		d.say(notice, "Error opening %s file: %s", rcFile, err)
+		os.Exit(1)
+	}
+	defer func(file *os.File) {
+		_ = file.Close()
+	}(file)
+
+	// Variables to be added or updated
+	exportLines := map[string]string{
+		"CURIO_DB_HOST":     fmt.Sprintf("export CURIO_DB_HOST=%s", strings.Join(d.HarmonyCfg.Hosts, ",")),
+		"CURIO_DB_PORT":     fmt.Sprintf("export CURIO_DB_PORT=%s", d.HarmonyCfg.Port),
+		"CURIO_DB_NAME":     fmt.Sprintf("export CURIO_DB_HOST=%s", d.HarmonyCfg.Database),
+		"CURIO_DB_USER":     fmt.Sprintf("export CURIO_DB_HOST=%s", d.HarmonyCfg.Username),
+		"CURIO_DB_PASSWORD": fmt.Sprintf("export CURIO_DB_HOST=%s", d.HarmonyCfg.Password),
+	}
+
+	// Flags to track whether we need to append these lines
+	existingVars := map[string]bool{
+		"CURIO_DB_HOST":     false,
+		"CURIO_DB_PORT":     false,
+		"CURIO_DB_NAME":     false,
+		"CURIO_DB_USER":     false,
+		"CURIO_DB_PASSWORD": false,
+	}
+
+	var lines []string
+	scanner := bufio.NewScanner(file)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		modified := false
+
+		// Check each export line to see if it exists and is not commented out
+		for key, newValue := range exportLines {
+			if strings.HasPrefix(line, "export "+key+"=") && !strings.HasPrefix(strings.TrimSpace(line), "#") {
+				lines = append(lines, newValue)
+				existingVars[key] = true
+				modified = true
+				break
+			}
+		}
+
+		// If no modifications were made, retain the original line
+		if !modified {
+			lines = append(lines, line)
+		}
+	}
 
-	// Write the file
-	err = os.WriteFile(where, []byte(strings.Join(args, "\n")), 0644)
+	if err := scanner.Err(); err != nil {
+		d.say(notice, "Error reading %s file: %s", rcFile, err)
+		os.Exit(1)
+	}
+
+	// Append missing export lines
+	for key, added := range existingVars {
+		if !added {
+			lines = append(lines, exportLines[key])
+		}
+	}
+
+	// Reopen the file in write mode to overwrite with updated content
+	file, err = os.OpenFile(rcFilePath, os.O_WRONLY|os.O_TRUNC, 0644)
```

Comment:
We should close the previous instance of `file` first

Never open critical system files with O_TRUNC, that's a recipe for disaster.

I'd open in append mode, append a newline, then our entries.

Reply:
That was a disaster waiting to happen. Forgot to replace the o_trunc with o_append